web/src/controllers/InstallerAPI.ts

```diff
@@ -261,3 +231,64 @@ export class Installers {
     return "";
   }
 }
+
+async function doMakeInstaller( response: Express.Response, request: Express.Request, id: string, slug: string) {
+  const auth = request.header("Authorization");
+  if (!auth) {
+    response.status(401);
+    return unauthenticatedResponse;
+  }
+
+  let teamID: string;
+  try {
+    teamID = await decode(auth);
+  } catch(error) {
+    response.status(401);
+    return unauthenticatedResponse;
+  }
+
+  if (!teamID) {
+    response.status(401);
+    return unauthenticatedResponse;
+  }
+
+  if (Installer.isSHA(id)) {
+    response.status(400);
+    return teamWithGeneratedIDResponse;
+  }
+  if (!Installer.isValidSlug(id)) {
+    response.status(400);
+    return slugCharactersResponse;
+  }
+  if (Installer.slugIsReserved(id)) {
+    response.status(400);
+    return slugReservedResponse;
+  }
+
+  let i: Installer;
+  try {
+    i = Installer.parse(request.body, teamID);
+  } catch(error) {
+    response.status(400);
+    return { error };
+  }
+  i.id = id;
+
+  if (i.spec.kotsadm && (i.spec.kotsadm.applicationSlug == undefined || i.spec.kotsadm.applicationSlug == "")) {
```

Comment:
```suggestion
  if (i.spec.kotsadm && !i.spec.kotsadm.applicationSlug) {
```

Reply:
I'll use this if I go this way instead of fixing things upstream